src/Workspaces/Core/Portable/Workspace/Solution/LoadTextOptions.cs

```diff
@@ -0,0 +1,38 @@
+﻿
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis;
+
+/// <summary>
+/// Options used to load <see cref="SourceText"/>.
+/// </summary>
+public readonly struct LoadTextOptions : IEquatable<LoadTextOptions>
```

Comment:
Is there a particular reason not to use a record struct here?

Reply:
> Is there a particular reason not to use a record struct here?

Yes, it generates a lot of things that we don't want

Reply:
I'm not sure it does here. We seem to be very intentionally using all those things.